### DISCUSSION TITLE
Using a dedicated search engine for filters

### DISCUSSION THREAD

**Discussion by code-runner-2017** [laravel/nova-issues#4982] [Q&A]
I was able to configure Laravel Scout for Nova Searches, as explained in the manual.
However, filters still use database queries. This can become inefficient when the number of records is high and the user wants to combine multiple filters. We solved this issue in other non-Nova applications using a separated search engine, such as ElasticSearch, for filters.

Assuming I could build completely custom filters for the purpose, is it possible to override the current resource filter implementation so that I can filter resources using, for instance, Solr and not the database?

**Comment by crynobone** [comment]
Can you share what you have done right now?

**Reply by code-runner-2017** [reply]
Are you asking about the Laravel Scout for Nova Searches or the non-Nova applications?

**Reply by crynobone** [reply]
> However, filters still use database queries. This can become inefficient when the number of records is high and the user wants to combine multiple filters

I would like to understand this claim, so I need to see what you have coded so far and how does resolve to "still use database queries"

**Reply by code-runner-2017** [reply]
Here are my steps:

- installed Laravel Scout (`composer require laravel/scout`)
- installed and configured this Scout driver for Apache Solr: https://github.com/Klaasie/laravel-scout-solr-engine
- created a Candidate Eloquent/Nova resource for testing
- added Laravel\Scout\Searchable to the Candidate model
- created some candidates and imported into Solr using Scout import
- checked that the candidates have been indexed in Solr

Everything was quite smooth, but:

- when I search for a candidate (Nova search on top of the listing table), it uses Solr. To be sure, I double checked removing the records from Solr (no results) and used Laravel Debugbar.
- when I filter candidates by last name, Solr is not involved. Debugbar shows that the query is a SQL query.

I'm using filterable fields (not custom fields).
I've also checked with the debugger.

It does not seem to be a bug, but a design choice. I can understand that it could be difficult for Nova to provide filters that work with both a DBMS and an external search engine. 

It would be fine with me to build a custom implementation.
This is why I'm asking if I can completely override the way filters are combined to build a query.
I wish I could somehow build my own Scout or Solr query using the search engine and just return a collection of models to Nova.
